x/sync/README.md

```diff
@@ -1,3 +1,59 @@
+4. `SyncGetChangeProofResponse`
+
+These message types are defined in `avalanchego/proto/sync.proto`.
+For more information on range proofs and change proofs, see their definitions in `avalanchego/merkledb/proof.go`.
+
+### `SyncGetRangeProofRequest`
+
+This message is sent from the client to the server to request a range proof for a given key range and root ID. That is, the client says, "Give me the key-value pairs that were in this key range when the database had this root."
+
+### `RangeProof`
+
+This message is sent from the server to the client in response to a `SyncGetRangeProofRequest`. It contains the key-value pairs that were in the requested key range when the database had the requested root, as well as a proof that the key-value pairs are correct.
+
+### `SyncGetChangeProofRequest`
+
+This message is sent from the client to the server to request a change proof between the given root IDs. That is, the client says, "Give me the key-value pairs that changed between the time the database had this root and that root."
+
+### `SyncGetChangeProofResponse`
+
+This message is sent from the server to the client in response to a `SyncGetChangeProofRequest`. If the server had sufficient history to generate a change proof, it contains a change proof that contains the key-value pairs that changed between the requested roots. If the server did not have sufficient history to generate a change proof, it contains a range proof that contains the key-value pairs that were in the database when the database had the latter root.
+
+## Algorithm
+
+For each key range, the sync client keeps track of the root ID of the database revision for which it has downloaded that key range. For example, it will store information that says something like, "I have all of the key-value pairs that were in range [`start`, `end`] when the database's root was `rootID`" for some keys `start` and `end`, and some database `rootID`. Note that `rootID` is the root ID that the client is trying to sync to, not the root ID of its own incomplete database.
+
+When the client starts syncing, it requests from a server a range proof for the entire database. The server replies with a range proof, which the client verifies. If it's valid, the key-value pairs in the proof are written to the database. If it's not, the client drops the proof and requests the proof from another server. 
+
+A range proof sent by a server may be valid but not contain all of the key-value pairs in the requested range. For example, a client might request all the key-value pairs in [`start`, `end`] but only receive those in range [`start`, `end'`] where `end'` < `end`. There might be too many key-value pairs to include in one message, or the server may be too busy to provide any more in its response. Unless the database is very small, this means that the range proof the client receives in response to its range proof request for the entire database will not contain all of the key-value pairs in the database.
```

Comment:
We need to write somewhere the word `continuous`.

> A range proof sent by a server may be valid but not contain all of the key-value pairs in the requested range.

sounds wrong.

```
A range proof sent by a server must return a continuous range of the key-value pairs, but may not return the full range that was requested.
```

Would be more clear imo

Reply:
I included the bit about validity since it's possible for the server to send the client an invalid range proof since it's a byzantine environment. Edited to mention that the client includes the maximum number of key-value pairs and maximum response size in its request. 

> Can you describe the current algorithm for how large of a Range Proof it currently returns? Is it complicated?

This is an implementation detail, so I'm not sure it makes sense to include it here, but basically the server tries to respond with the entire key-value range, but if the proof ends up being too large, it includes only a portion of the range.